pages/user/PKPUserHandler.inc.php

```diff
@@ -50,16 +51,26 @@ public function setLocale($args, $request)
             $session->setSessionVar('currentLocale', $setLocale);
         }
 
-        $source = $request->getUserVar('source');
-        if (preg_match('#^/\w#', $source) === 1) {
-            $request->redirectUrl($source);
+        if (!isset($_SERVER['HTTP_REFERER'])) {
```

Comment:
I didn't watch the PR (I mean the problem vs solution), so this is just a generic comment about the `HTTP_REFERER` 😁

Given this header might be removed by enterprise proxies and I remember Opera (or Firefox) also had an option to disable it, I think it would be better to use the `Origin` header, which should be safer (if proxies attempt to break it, they will also break CORS requests), except for really old browsers.